markdown/events/2021/bytesize-12-template-sync.md

```diff
@@ -38,3 +38,94 @@ The talk will be presented on Zoom and live-streamed on YouTube:
 * YouTube: <https://youtu.be/-CZKoo5Y_08>
 * Bilibili: <https://www.bilibili.com/video/BV1D64y127yJ/>
 * FigShare: <https://doi.org/10.6084/m9.figshare.14572866.v1>
+
+<details markdown="1"><summary>Video transcription</summary>
+**Note: The content has been edited to make it reader-friendly**
+
+[0:46](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=46) Hi everyone, I’ll be going over how template synchronization works. This talk is targeted at pipeline developers within the nf core ecosystem and for those who are interested in building their own pipeline based on the nf-core toolsets.
+
+[1:03](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=63) First, a refresher here. All of this is based very heavily on Git and GitHub, so let’s very quickly go over some of what we’ve covered before [Bytesize#4](https://nf-co.re/events/2021/bytesize-4-github-contribution-basics). Git is a version control code system that has a bit of jargon that you need to know. You do `commit`, switching little packages of work so you do some edits and then bubble them together and then say `commit` and that’s like checkpoints in the timeline of your code. They go in a kind of straight line, and then you can make branches, where you can work on different things at the same time.
+
+[1:44](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=104) You can carry on working in that branch.
+
+[1:47](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=108) And when you want to, you can merge that branch back into the main line of the commit history.
+
+[1:55](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=115) You can fork an entire repository, which basically copies a repository of code into your account. So for example, I have forked an nf-core pipeline to my personal Github account.
+
+[2:08](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=128) And you can carry on doing development on your fork. When you want to contribute what you’ve done back to the main repository within nf-core, you open up a `pull request` that is then merged into the main line of commits. Then everything you’ve worked on becomes a part of the main repository.
+
+[2:29](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=150) All our nf core pipelines have at least three branches called `dev`, `master` and `TEMPLATE`. The `master` branch is for default (after the first release), and should only ever have the latest stable release code. That is so that if someone runs Nextflow with the pipeline name, but does not specify a release, it can grab the default branch, which has the latest stable release.
```

Comment:
```suggestion
[2:29](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=150) All our nf-core pipelines have at least three branches called `dev`, `master` and `TEMPLATE`. The `master` branch is for default (after the first release), and should only ever have the latest stable release code. That is so that if someone runs Nextflow with the pipeline name, but does not specify a release, it can grab the default branch, which has the latest stable release.
```